virtool/jobs/build_index.py

```diff
@@ -154,6 +157,59 @@ async def upload(job):
         })
 
 
+async def build_json(job):
+    """
+    Create a reference.json.gz file at ``<data_path>/references/<ref_id>/<index_id>``.
+
+    """
+    await job.db.indexes.find_one_and_update({"_id": job.params["index_id"]}, {
+        "$set": {
+            "has_json": False
+        }
+    })
+
+    document = await job.db.references.find_one(job.params["ref_id"], ["data_type", "organism", "targets"])
+
+    app_dict = {
+        "db": job.db,
+        "settings": job.settings
+    }
+
+    otu_list = await virtool.references.db.export(
```

Comment:
Change signature of `virtool.references.db.export` to `def export(db, settings, ref_id)` to avoid creating `app_dict`.

Update usage in `virtool.downloads.api.download_reference`.

Reply:
I think if `export` function has changed, we also need to change the signature of `virtool.history.db.patch_to_version` because it has been called in `export` function and it also needs the `app` parameter.

Do you want this to be changed as well?

Reply:
Ah. What a pain. Leave it as is then.